### PR TITLE
Feat(now don't need to add so much info for adding a commit): 

### LINKED ISSUE #447: CI(semver): Add semantic versioning
## Problem
We currently use semantic versioning but rely on developers to manually update versionings and change logs. This creates lots of room for error.

## Solution
Enforce proper semantic versioning and change log updates via [commitizen](https://www.npmjs.com/package/commitizen).

### LINKED ISSUE #477: CI(commitizen): Implement Conventional Commit Logs
## Problem
Many automated versioning tools such as semantic versioning rely on machine-readable commit logs to determine which versions of a library to bump. In doing so, they assume commit logs follow the [conventional commit style](https://www.conventionalcommits.org/en/v1.0.0/).

We have not historically followed this conventional nor do we have any way to enforce it which blocks us from automating our versioning using semver (assuming this is our desired approach).

## Solution
Use [committizen](https://www.npmjs.com/package/commitizen).

### PR DESCRIPTION
**Pull request by Suvendu-UI**
implementted commitizen easy to commit

fixed #477 
Screenshot 2024-07-07 214606
Screenshot 2024-07-07 214619
Screenshot 2024-07-07 214632

### CODE REVIEW

**Comment by EandrewJones** [comment]
Hi @Suvendu-UI, thanks for the PR and your interest in the project! We always welcome contributions.

I see this commit is for #447. We encourage discussion prior to opening PRs to help ensure alignment. Per the comments, we felt this issue needs more discussion before acting on a solution:
> We do not necessarily want to manage version updates at the commit level. Is there a way to do this via commitizen or do we need an alternative?

I would love for you to work with us to find the right solution and help shepherd this PR through. I'll tag you in further discussion on the issue.

**Comment by Suvendu-UI** [comment]
Hey, @EandrewJones , how are you, i was looking for easier ways to commit the changes without much hecticness involved , i was googling it and got a automated ai which commits instantly  which is almost equiv. to commitizen.

Here is the link https://github.com/christian-gama/autocommit

**Comment by EandrewJones** [comment]
Hi Suvenda,

Flagon is very particular about its dependency footprint. We try to keep
the tree as small as possible and only use well established tools.
Committizen passes that bar, autocommit does not.

Moreover, we believe contributors should write their own commit messages.
Commit messages should be concise, relevant, clear, and consistent in
style. There's no need to try to add AI (extra complexity) for something as
simple as commits.

Best

Evan Jones
Website: www.ea-jones.com

File: .czrc [hunk 1]
```diff
@@ -0,0 +1 @@
+{ "path": "cz-conventional-changelog" }
```
Comment (Jyyjy) [inline .czrc:1]: This file may not be needed

**Review by Jyyjy — CHANGES_REQUESTED** [review]
A  couple things.
1. For the GitHub action pipeline to succeed, add the new .czrc file to the ignored paths in .licenserc.yaml. Every file that can have a APL 2.0 license header must. In this case, .czrc is JSON, so it cannot have a header and should be ignored.
2. This project uses [pre-commit hooks](https://pre-commit.com/) to minimize surprise failures from GitHub actions. Please install.
3. Ignore the first point because I don't think .czrc is needed. Based on the [docs](https://github.com/commitizen/cz-cli?tab=readme-ov-file#making-your-repo-commitizen-friendly), you should define the commitizen config either in package.json or .czrc. Package.json is preferred to limit root dir clutter.

Thanks for taking interest in our project. Out of curiosity, are you a Google summer of code participant?

File: .czrc (hunk 1 above)
Comment (Suvendu-UI) [reply .czrc:1]: thanks for the suggestion

File: build/userale-2.4.0.min.js [hunk 2]
```diff
@@ -15,4 +15,783 @@
...
+!(function (e, t) {
```
Comment (Jyyjy) [inline build/userale-2.4.0.min.js:18]: Something is not correct with your build. This file should be single line, if you run `npm install` followedd by `npm build` is this what you get?

File: package-lock.json
Comment (Jyyjy) [inline package-lock.json:1]: You'll have to rebase against master to be able to merge. The conflicts are coming from this file which is auto generated. Don't bother fixing it within the rebase. Instead, pick one then re generate it after the rebase

https://git-scm.com/book/en/v2/Git-Branching-Rebasing

**Review by Jyyjy — CHANGES_REQUESTED** [review]

File: build/userale-2.4.0.min.js (hunk 2 above)
Comment (Suvendu-UI) [reply build/userale-2.4.0.min.js:18]: hey, first when i saw this code i thought it was a mistake so i formatted it using prettier to have the proper indentation, anyways its not a problem, its the same code. I will revert the changes

**Comment by Suvendu-UI** [comment]
Hey @Jyyjy i think i have resolved the requested changes

**Comment by EandrewJones** [comment]
@Suvendu-UI There's a tiny conflict that needs to be resolved before we can merge this. You should be able to:
1. rebase this branch on top of the latest changes from `master`
2. select either `package-lock.json` file, i.e. `git checkout --ours package-lock.json`
3. reinstall, build, and push

Pinging @Jyyjy for his final look over.

**Comment by Suvendu-UI** [comment]
done all the steps @EandrewJones  and @Jyyjy

**Comment by EandrewJones** [comment]
@Suvendu-UI Are you sure? There's still a conflict on package-lock.json.

Are you sure you've pulled down the most up-to-date changes from the upstream repo, i.e. `apache/flagon-useralejs` as opposed to just your fork (`Suvendu-UI/flagon-useralejsGSOC`)?

I see you're working directly off master on your fork (not recommended, but okay for now). That changes the steps slightly.

To be more explicit:
1. `git checkout master`
2. `git pull upstream master` where upstream points to the apache repo
3. `git push origin master` where origin points to your fork
4. This will raise a conflict in your `package-lock.json`. You should be able to resolve it by simply deleting the file, then re-installing `npm install`.
5. Commit your changes and push them to your fork.

**Comment by Suvendu-UI** [comment]
now all resolved i guess @EandrewJones

**Comment by Suvendu-UI** [comment]
i don't know why but i am unable to pass the  npm ci error , i have tried all the possible method reinstalling the package-lock and package file fsversion , can you help me?

**Comment by EandrewJones** [comment]
> i don't know why but i am unable to pass the npm ci error , i have tried all the possible method reinstalling the package-lock and package file fsversion , can you help me?

I just got a chance to look into this. I've never seen this issue before :exploding_head: 

Let's try a few things:
1. Can you share what node  and npm versions you're using? Please ensure you're using a Node version >= 18.x.x. If you're not already using NVM, I recommend it (see [here](https://stackoverflow.com/questions/76387479/npm-err-npm-ci-can-only-install-packages-when-your-package-json-and-package-l) for more details).
2. Check the suggested fixes [here](https://stackoverflow.com/questions/76387479/npm-err-npm-ci-can-only-install-packages-when-your-package-json-and-package-l)

You may need to delete the `node_modules` and `package-lock.json`, then using correct node version, try running `npm i && npm ci`.

**Pull request closed without merging**